### PR TITLE
[SPARK-50167][PYTHON][CONNECT] Improve PySpark plotting error messages and imports

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Improve PySpark plotting error messages and imports:
- Minor improvements to error messages for better clarity.
- Delayed imports of PySparkPlotAccessor as a best practice to avoid potential circular imports in the future.

### Why are the changes needed?
Improving error messages enhances usability, while delaying PySparkPlotAccessor imports prevents potential circular dependencies.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Unit tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: python/pyspark/sql/classic/dataframe.py [hunk 1]
```diff
@@ -1795,7 +1791,9 @@ def executionInfo(self) -> Optional["ExecutionInfo"]:
         )
 
     @property
-    def plot(self) -> PySparkPlotAccessor:
+    def plot(self) -> "PySparkPlotAccessor":
```
Comment (zhengruifeng) [inline python/pyspark/sql/classic/dataframe.py:1794]: nit: we may need `# type: ignore[name-defined] # noqa: F821` :)

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/classic/dataframe.py (hunk 1 above)
Comment (xinrong-meng) [reply python/pyspark/sql/classic/dataframe.py]: Good catch! Thanks :)

File: python/pyspark/errors/error-conditions.json [hunk 2]
```diff
@@ -817,12 +817,12 @@
       "Pipe function `<func_name>` exited with error code <error_code>."
     ]
   },
-    "PLOT_INVALID_TYPE_COLUMN": {
+  "PLOT_INVALID_TYPE_COLUMN": {
     "message": [
       "Column <col_name> must be one of <valid_types> for plotting, got <col_type>."
     ]
   },
-  "PLOT_NOT_NUMERIC_COLUMN": {
```
[the remark this answers is not in the document]
Comment (xinrong-meng) [reply python/pyspark/errors/error-conditions.json:825]: PLOT_NOT_NUMERIC_COLUMN_ARGUMENT focuses on **argument** must be numeric.
PLOT_INVALID_TYPE_COLUMN indicates that a specific **column** must be of a valid type.
How about we keep both for clarity?

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @xinrong-meng and @zhengruifeng for review.

**Comment by xinrong-meng** [comment]
Thank you @MaxGekk ! : )

**Pull request closed without merging**